pages/workspaces/[workspaceId]/contributor-insights/index.tsx

```diff
@@ -33,7 +35,9 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
   });
 
   if (error) {
-    if (error.status === 404) {
+    deleteCookie(context.res, WORKSPACE_ID_COOKIE_NAME);
+
+    if (error.status === 404 || error.status === 401) {
       return { notFound: true };
     }
```

Comment:
not sure if it's worth it, but maybe wrapping up this same logic in all these pages into a function might make sense. Not for this PR though.